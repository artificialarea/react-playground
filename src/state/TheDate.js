import React from 'react';

class TheDate extends React.Component {

  constructor(props) {
    super(props)
    this.state = {
      datetime: new Date(),
      foo: 'foo',
      hello: 'Hello, world!',
      list: [1, 2, 3],
      obj: { nested: 'object', yes: true },
    }
  }

  render() {
    console.log(this.state);
    return <div>{this.state.datetime.toLocaleString()}</div>
  }
}

export default TheDate;