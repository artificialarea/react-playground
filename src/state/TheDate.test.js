import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TheDate from './TheDate';

describe('<TheDate />', () => {

    it('render without crashing', () => {
      const div = document.createElement('div')
      ReactDOM.render(<TheDate />, div)
      ReactDOM.unmountComponentAtNode(div);
    });

    // 'it.skip' work-in-progress
    // we know this test in it's current form won't work as the snapshots of the ever-changing time will never match, so telling jest to 'skip' the snapshot for now...
    it.skip('renders UI without crashing', () => {
      expect(
        renderer.create(<TheDate />).toJSON()
      ).toMatchSnapshot()
    });

});