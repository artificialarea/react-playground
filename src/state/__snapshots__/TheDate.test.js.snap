// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TheDate /> renders UI without crashing 1`] = `
<div>
  5/13/2020, 11:54:51 AM
</div>
`;
